src/modalities/dataloader/create_packed_data.py

```diff
@@ -62,11 +63,11 @@ def __init__(
         self.tokenizer = tokenizer
         self.eod_token = eod_token
         self._token_size_in_bytes = self._get_required_num_of_bytes_to_repr(self.tokenizer.vocab_size)
-        encoded_eod_token = self.tokenizer.get_token_id(self.eod_token)
-        self._encoded_eos_token_as_bytes = self._encoded_token_to_bytes(encoded_eod_token)
+        eod_token_id = self.tokenizer.get_token_id(self.eod_token)
```

Comment:
What happens if the provided `eod_token` does not result into a singke token id?

Reply:
In the case of the the gpt2 tokenizer, the unk_token is returned which is equivalent to the eod token (<|endoftext|>). 
The warning informs the user of the potentially problematic equivalence: 
https://github.com/Modalities/modalities/blob/85f13acb282527efc23d2ebf4548d0c2f3200726/src/modalities/tokenization/tokenizer_wrapper.py#L185-L186

I added another check, that makes sure that we receive a single integer value.